pkg/controller/servicebinding/servicebinder.go

```diff
@@ -273,14 +279,7 @@ func removeFinalizer(sbr *v1alpha1.ServiceBinding) {
 // handleApplicationError handles scenarios when:
 // 1. application not declared in the Service Binding
 // 2. application not found
-func (b *serviceBinder) handleApplicationError(reason string, applicationError error, sbrStatus *v1alpha1.ServiceBindingStatus) (reconcile.Result, error) {
```

Comment:
@Avni-Sharma 
Put the condition setting out of handleApplicationError function as there are two callers:
caller1: https://github.com/redhat-developer/service-binding-operator/blob/2f561b7bb5cecb0e2d497d3bff352b1edf578ff6/pkg/controller/servicebinding/servicebinder.go#L327
caller2: https://github.com/redhat-developer/service-binding-operator/blob/2f561b7bb5cecb0e2d497d3bff352b1edf578ff6/pkg/controller/servicebinding/servicebinder.go#L343

caller1 is for application empty so InjectionReady is false and BindingReady is true.
caller2 is for application not found so both InjectionReady and BindingReady are false.
So here we need to put the condition setting out of this function.

Reply:
Thanks. However, in order to improve readability of the code, and avoid copy/pasting `conditionsv1.SetStatusCondition` blocks around, how about to define `InjectionReadyStatus(...)`, `ReadyStatus`, ... on `ServiceBindingStatus` struct and use them across the code? IMHO, it would improve readbility and reusability of the code. Also, Ready condition could even autoderived from other two conditions in that case.